src/es_archiver/lib/indices/kibana_index.ts

```diff
@@ -93,61 +61,21 @@ export async function deleteKibanaIndices({
  */
 export async function migrateKibanaIndex({
   client,
-  log,
-  kibanaPluginIds,
+  kbnClient,
 }: {
   client: Client;
-  log: ToolingLog;
-  kibanaPluginIds: string[];
+  kbnClient: KbnClient;
 }) {
-  const uiExports = await getUiExports(kibanaPluginIds);
-  const kibanaVersion = await loadKibanaVersion();
-
-  const configKeys: Record<string, string> = {
-    'xpack.task_manager.index': '.kibana_task_manager',
-  };
-  const config = { get: (path: string) => configKeys[path] };
-
-  const savedObjectTypes = convertLegacyTypes(uiExports, config as LegacyConfig);
-  const typeRegistry = new SavedObjectTypeRegistry();
-  savedObjectTypes.forEach(type => typeRegistry.registerType(type));
-
-  const logger = {
-    trace: log.verbose.bind(log),
-    debug: log.debug.bind(log),
-    info: log.info.bind(log),
-    warn: log.warning.bind(log),
-    error: log.error.bind(log),
-    fatal: log.error.bind(log),
-    log: (entry: any) => log.info(entry.message),
-    get: () => logger,
-  };
-
-  const migratorOptions = {
-    savedObjectsConfig: {
-      scrollDuration: '5m',
-      batchSize: 100,
-      pollInterval: 100,
-      skip: false,
+  // we allow dynamic mappings on the index, as some interceptors are accessing documents before
+  // the migration is actually performed. The migrator will put the value back to `strict` after migration.
+  await client.indices.putMapping({
+    index: '.kibana',
+    body: {
+      dynamic: true,
```

Comment:
As explained in comment, I could not find which one exactly, but some interceptors are accessing / creating a `config` object during interception (so before the migrator actually created the correct mappings for the index). As the mapping is set to `strict`, this cause an unrecoverable error

```
'mapping set to strict, dynamic introduction of [references] within [_doc] is not allowed: [strict_dynamic_mapping_exception] mapping set to strict, dynamic introduction of [references] within [_doc] is not allowed'
```

We change the type to `dynamic` before calling the migrator. It will be changed back to `strict` by the migrator itself.

Reply:
> It will be changed back to strict by the migrator itself.

I'm ok with this workaround, but do we have actual tests that verify that the mappings are set to strict? Just want to make sure that assumption doesn't get violated by accident one day.

Reply:
We got unit tests covering that in `src/core/server/saved_objects/migrations/core/build_active_mappings.test.ts`. I can add an additional test testing explicitly that in this test suite.

I think this is sufficient, but tell me if we want to add an actual FTR test on that (this would be quite hard to test in FTR though, not sure how we could achieve that)

Reply:
Added a specific unit test